dev/lib/product_taxonomy/models/attribute.rb

```diff
@@ -2,15 +2,101 @@
 
 module ProductTaxonomy
   class Attribute
-    attr_reader :id, :name, :description, :friendly_id, :handle, :values
+    include ActiveModel::Validations
 
-    def initialize(id:, name:, description:, friendly_id:, handle:, values:)
+    class << self
+      # Load attributes from source data. By default, this data is deserialized from a YAML file in the `data` directory.
+      #
+      # @param source_data [Array<Hash>] The source data to load attributes from.
+      # @param values [Hash<String, Value>] A hash of {Value} objects keyed by their friendly ID.
+      # @return [ModelIndex<Attribute>] A model index of {Attribute} objects.
+      def load_from_source(source_data:, values:)
+        model_index = ModelIndex.new(self, hashed_by: :friendly_id)
+        raise ArgumentError, "source_data must be a hash" unless source_data.is_a?(Hash)
+        raise ArgumentError, "source_data must contain keys \"base_attributes\" and \"extended_attributes\"" unless
+          source_data.keys.sort == ["base_attributes", "extended_attributes"]
+
+        ["base_attributes", "extended_attributes"].each do |type|
+          load_attributes(type, source_data, model_index, values)
+        end
```

Comment:
Why this vs a more plain:
```rb
load_attributes("base_attributes", source_data, model_index, values)
load_attributes("extended_attributes", source_data, model_index, values)
```

Reply:
Agreed, on second thought the block is unnecessary here